goldens/public-api/manage.js

```diff
@@ -0,0 +1,54 @@
+const {exec} = require('shelljs');
+const minimist = require('minimist');
+
+// Remove all command line flags from the arguments.
+const argv = minimist(process.argv.slice(2));
+// The command the user would like to run, either 'accept' or 'test'
+const USER_COMMAND = argv._[0];
+// The shell command to query for all Public API guard tests.
+const BAZEL_PUBLIC_API_TARGET_QUERY_CMD =
+    `yarn -s bazel query --output label 'kind(nodejs_test, ...) intersect attr("tags", "api_guard", ...)'`
+// Bazel targets for testing Public API goldens
+process.stdout.write('Gathering all Public API targets');
+const ALL_PUBLIC_API_TESTS = exec(BAZEL_PUBLIC_API_TARGET_QUERY_CMD, {silent: true})
+                                 .trim()
+                                 .split('\n')
+                                 .map(test => test.trim());
+process.stdout.clearLine();
+process.stdout.cursorTo(0);
```

Comment:
I had issues in the past with this script in the FW repo (on windows), which is why I always used a custom bash script with `xargs`. The bazel query above never returns anything:

```
Failed run: .accept
  Loading:
  Loading: 0 packages loaded
  ERROR: Skipping '.accept': no such target '//:.accept': target '.accept' not declared in package '' defined by C:/users/paul/projects/angular/BUILD.bazel
  WARNING: Target pattern parsing failed.
  ERROR: no such target '//:.accept': target '.accept' not declared in package '' defined by C:/users/paul/projects/angular/BUILD.bazel
  INFO: Elapsed time: 0.051s
```

The `ALL_PUBLIC_API_TESTS` array just was an array with an empty string as single item FWIW.

Reply:
I actually wanted to investigate adding a more generalized version of this into the dev-infra tools directly since variations of this are done in several places.